### PR TITLE
WordPress.com Toolbar: Add View Site item to My Sites menu

### PR DESCRIPTION
**Pull request by vindl**
Syncing the Masterbar with the Calypso version that now has View Site (for Jetpack sites) item bellow Blog info.

Fixes https://github.com/Automattic/jetpack/issues/6877
Depends on: D6074-code

#### Testing instructions:

WP.com side (needed for styles):

* Apply D6074-code to your sandbox.
* Sandbox `s0.wp.com`, `s1.wp.com`, and `s2.wp.com`.

Jetpack side: 

* Open up `wp-admin` of your test Jetpack site and verify that Site Preview item is present in My Sites menu.
* View the same menu on site's front end and verify that Site Preview is not present.

### Visual changes

Before:

before

After: 

before

### Proposed changelog
Masterbar: added a View Site item to My Sites menu that launches the front end of the current site.

### CODE REVIEW

**Comment by rickybanister** [comment]
Just to make sure I'm on the same page—why is this link necessary for the front-end masterbar? If the front-end bar is shown when looking at the site, what would the site preview do?

**Comment by roundhill** [comment]
Hmm I don't see the site preview link when in `wp-admin` on my site on .com:

screen shot 2017-06-21 at 10 57 12 am

Are you adding it there as well?

**Comment by vindl** [comment]
> Just to make sure I'm on the same page—why is this link necessary for the front-end masterbar? If the front-end bar is shown when looking at the site, what would the site preview do?

@rickybanister It won't be shown on the front-end, only when someone is viewing the Masterbar in `wp-admin` (more info in https://github.com/Automattic/jetpack/issues/6877).

> Hmm I don't see the site preview link when in wp-admin on my site on .com. Are you adding it there as well?

@roundhill yes, D6074-code should handle that.

File: modules/masterbar/masterbar.php [hunk 1]
```diff
@@ -563,6 +563,20 @@ public function add_my_sites_submenu( $wp_admin_bar ) {
 			) );
 		}
 
+		// Site Preview
+		if ( is_admin() ) {
+			$wp_admin_bar->add_menu( array(
+				'parent' => 'blog',
+				'id'     => 'site-preview',
+				'title'  => __( 'Site Preview' ),
```
Comment (eliorivero) [inline modules/masterbar/masterbar.php]: This should be
```
'title'  => esc_html__( 'Site Preview', 'jetpack' ),
```

**Comment by eliorivero** [comment]
This works fine. After the issue is addressed, this is good to go.

File: modules/masterbar/masterbar.php (hunk 1 above)
Comment (vindl) [inline modules/masterbar/masterbar.php]: Thanks @eliorivero, added it in d5e87c85a42dc764f3f3dbf401ee0c2e8688d286 (dropped `esc_html` since I don't think we need it here). I've also changed the title following the discussion in D6074-code.

Comment (eliorivero) [inline modules/masterbar/masterbar.php]: Stephane puts it in eloquent words here
https://core.trac.wordpress.org/ticket/30724#comment:12

> One problem with not escaping translations is that some plugins that filter translations will allow end users to push translations in the back end. 
Depending on where these go they can break the code because characters are not escaped properly and the end users doing the translations won't understand what broke.
This doesn't even take under account that you may not be able to trust the end users doing the translations.
I feel that escaping protects against user error and potentially malicious users especially with translation plugins with little cost / negative repercussions.
Thanks!

But anyway, don't want to block you for this so let's go ahead.

Comment (vindl) [inline modules/masterbar/masterbar.php]: Oh right, guarding against malicious translations and plugins that filter them totally makes sense. Since we should be doing that in several other places in Masterbar's code, I'm going to spin up a separate PR that fixes that. Thanks for pointing this out! :heart:

Comment (vindl) [inline modules/masterbar/masterbar.php]: Created a PR to fix this https://github.com/Automattic/jetpack/pull/7386.
